### ISSUE TITLE
Sign-in errors redirect to /error instead of /signin ?

### ISSUE THREAD

**Issue by balazsorban44** [nextauthjs/next-auth#3341]
TLDR;

As documented in https://next-auth.js.org/configuration/pages#sign-in-page

The `EmailSignin` error should redirect to the sign-in page, both in case of the built-in and a custom page. Currently, it redirects to the error page.

The error message also seems to be wrong.

`EmailSignin` is shown when the e-mail wasn't sent.

This line doesn't make sense: https://github.com/nextauthjs/next-auth/blob/4dcdb62dcabad8cbc14a5357cf6c5690bb12c3d1/src/core/pages/signin.tsx#L51

It should say something like "The e-mail could not be sent."

### Discussed in https://github.com/nextauthjs/next-auth/discussions/3336

<div type='discussions-op-text'>

<sup>Originally posted by **ptrkc** December  2, 2021</sup>
Hello everyone. I am a bit confused about errors redirect.
I have a custom sign-in page, `/login`, and by reading [the docs](https://next-auth.js.org/configuration/pages#error-codes), I understand that Sign-in page errors should redirect to `/login?error=`.
I tried to sign-in with a broken Email provider and got redirected to `/login?error=EmailSignin`, good.
Now, I tried adding a custom error page, `/error`, to handle other errors like expired sign-in link, etc.
After adding the custom error page, I no longer get redirected to `/login?error=EmailSignin`, but to `/error?error=EmailSignin`.

Is this the expected behavior? I know I can use `redirect: false` and treat the errors on the sign-in page, but I think the documentation says sign-in page errors should redirect to itself.</div>

**Comment by klausbadelt** [comment]
This might not be a bug, just a documentation issue.

The way it works right now is, _all_ errors hit the errors page, which in turn redirects _some_ ([these 10](https://next-auth.js.org/configuration/pages#sign-in-page)) to the Sign-in page. Errors never hit the Sign-in page "directly".

Documentation should say that your custom error page should implement this redirect behavior.

**Comment by balazsorban44** [comment]
I don't think we should do double redirects if it's easy to avoid.
